grammar.js

```diff
@@ -1117,11 +1122,12 @@ module.exports = grammar(C, {
         }));
     },
 
-    argument_list: $ => seq(
+    // The compound_statement is added to parse macros taking statements as arguments, e.g. MYFORLOOP(1, 10, i, { foo(i); bar(i); })
+    argument_list: $ => prec('argument_list', seq(
       '(',
-      commaSep(choice($._expression, $.initializer_list)),
+      commaSep(choice(seq(optional('__extension__'), $._expression), $.initializer_list, $.compound_statement)),
```

Comment:
I'd argue it's decently commonly used, and adds nothing to the state count, and in general I'm inclined to agree to avoid caring about macro usages here, but this is popular enough and harmless enough to add imo.